### PR TITLE
nodenext compatibility

### PR DESCRIPTION
**Pull request by Uzlopak**
Part of [Migration of the infamous triplet to next level, Nodenext support ](https://github.com/fastify/fastify/issues/4349)

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
I would recommend to avoid using the git reference to the pem repo and just merge it even though the ci pipeline fails.

File: package.json [hunk 1]
```diff
@@ -42,10 +45,11 @@
     "mocha": "^10.0.0",
     "nock": "^13.1.3",
     "nyc": "^15.1.0",
-    "pem": "^1.14.4",
+    "pem": "Dexus/pem#master",
```
Comment (Eomm) [inline package.json:48]: Could not we use https://www.npmjs.com/package/self-cert ?

Comment (Uzlopak) [reply package.json:48]: Thought about it. I will just generate the rsa keys once. Then we dont need pem at all.

**Comment by Uzlopak** [comment]
Replaced pem with self-cert in #72.

So first we merge #72 and then we merge master into this branch, and then it should be good ;)

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json]: See comment below ;)

**Review by Eomm — APPROVED** [review]

**Pull request merged**